lymph/unilateral.py

```diff
@@ -73,6 +72,81 @@ def change_base(
     else:
         return pad + result[::-1]
 
+def check_modality(modality: str, spsn: list):
+    """Private method that checks whether all inserted values
+    are valid for a confusion matrix.
+
+    Args:
+        modality (str): name of the modality
+        spsn (list): list with specificity and sensiticity
+
+    Raises:
+        TypeError: returns a type error if the modality is not a string
+        ValueError: raises a value error if the spec or sens is not a number btw. 0.5 and 1.0
+    """
+    if not isinstance(modality, str):
+            msg = ("Modality names must be strings.")
+            raise TypeError(msg)
+    has_len_2 = len(spsn) == 2
+    is_above_lb = np.all(np.greater_equal(spsn, 0.5))
+    is_below_ub = np.all(np.less_equal(spsn, 1.))
+    if not has_len_2 or not is_above_lb or not is_below_ub:
+        msg = ("For each modality provide a list of two decimals "
+            "between 0.5 and 1.0 as specificity & sensitivity "
+            "respectively.")
+        raise ValueError(msg)
+
+def clinical(modality_spsn: Dict[Any, List[float]]) -> Dict:
+    """produces the confusion matrix of a clinical modality, i.e. a modality
+    that can not detect microscopic metastases
+
+    Args:
+        modality_spsn (Dict[Any, List[float]]): dictionary with modality name and [sp,sn] list
+
+    Returns:
+        Dict[str,np.ndarray]: returns a dictionary with modality name and clinical confusion matrices
+    """
+    modality_dictionary = {}
+    for mod, spsn in modality_spsn.items():
+        try:
+            check_modality(mod,spsn)
+        except TypeError:
+            raise
+        except ValueError:
+            raise
+
+        sp, sn = spsn
+        modality_dictionary[mod] = np.array([[sp, 1. - sp],
+                                            [sp, 1. - sp],
+                                            [1. - sn, sn]])
+    return modality_dictionary
+
+
+def pathological(modality_spsn: Dict[Any, List[float]]) -> Dict:
+    """produces the confusion matrix of a pathological modality, i.e. a modality
+    that can detect microscopic metastases
+
+    Args:
+        modality_spsn (Dict[Any, List[float]]):  dictionary with modality name and [sp,sn] list
+
+    Returns:
+        Dict[str,np.ndarray]: returns a dictionary with modality name and pathological confusion matrices
+    """
+    modality_dictionary = {}
+    for mod, spsn in modality_spsn.items():
+        try:
+            check_modality(mod,spsn)
+        except TypeError:
+            raise
+        except ValueError:
+            raise
+
+        sp, sn = spsn
+        modality_dictionary[mod] = np.array([[sp, 1. - sp],
+                                            [1. - sn, sn],
+                                            [1. - sn, sn]])
+    return modality_dictionary

```

Comment:
Actually you can still set them like this:

`model.modalities  = lymph.clinical(modality_spsn={'clinical_consensus': [0.86, 0.81],
                                'CT': [0.76, 0.81],
                                'MRI': [0.63, 0.81],
                                'PET': [0.86, 0.79],
                                'pCT': [0.86, 0.81]})

model.modalities = lymph.pathological({'pathology': [1, 1],
                                    'FNA': [0.98, 0.80]})`

Reply:
Actually you can still set several modalities like this:

```
model.modalities  = lymph.clinical(modality_spsn={'clinical_consensus': [0.86, 0.81],
                                'CT': [0.76, 0.81],
                                'MRI': [0.63, 0.81],
                                'PET': [0.86, 0.79],
                                'pCT': [0.86, 0.81]})

model.modalities = lymph.pathological({'pathology': [1, 1],
                                    'FNA': [0.98, 0.80]})
```
My idea was that you set the clinical and pathological modalities in one go, hence calling the function twice.
But I could also adapt it to be applicable as you mentioned it.

Reply:
Ah, I see. Yeah, I think so far it removed all modalities when new ones were set. And I do have the feeling it is slightly safer to implement it that way.

Maybe it would make sense if the `modalities` attribute were an instance of a `dict` subclass, similar to the marginalization thingy... Anyways, that's maybe for later.